Src/Scripts/Include-GenerateUnitClassSourceCode.ps1

```diff
@@ -278,11 +280,30 @@ namespace UnitsNet
         public static $className Parse(string str, IFormatProvider formatProvider = null)
         {
             if (str == null) throw new ArgumentNullException("str");
-            string[] words = str.Split(new[] {" "}, StringSplitOptions.RemoveEmptyEntries);
-            if (words.Length < 2)
+
+            var numFormat = formatProvider != null ?
+                (NumberFormatInfo) formatProvider.GetFormat(typeof (NumberFormatInfo)) :
+                (NumberFormatInfo) CultureInfo.CurrentCulture.NumberFormat.Clone();
+
+            var numRegex = @"[\d., "                        // allows digits, dots, commas, and spaces in the number by default
+                         + numFormat.NumberGroupSeparator   // adds provided (or current) culture's group separator
+                         + numFormat.NumberDecimalSeparator // adds provided (or current) culture's group separator
+                         + @"]*\d";                         // ensures quantity ends in digit
+            var regexString = @"\s*"                                                             // ignore leading whitespace
+                            + @"(?<value>[-+]?" + numRegex + @"(?:[eE][-+]?" + numRegex + @")?)" // capture Quantity input
+                            + @"\s?"                                                             // ignore whitespace (if any)
+                            + @"(?<unit>\S+)"                                                    // capture Unit (non-whitespace) input
+                            + @"\s*";                                                            // ignore trailing whitespace
+            var regex = new Regex(regexString);
+            GroupCollection groups = regex.Match(str).Groups;
+
+            var valueString = groups["value"].Value;
+            var unitString = groups["unit"].Value;
+
+            if (valueString == "" || unitString == "")
             {
                 var ex = new ArgumentException(
-                    "Expected two or more words. Input string needs to be in the format \"<quantity> <unit>\".", "str");
+                    "Expected valid quantity and unit. Input string needs to be in the format \"<quantity> <unit>\".", "str");
```

Comment:
Add info about second format option with no whitespace between quantity and unit.